whimsical-woodpeckers/messages/consumers.py

```diff
@@ -0,0 +1,80 @@
+import json
+from channels.generic.websocket import WebsocketConsumer
+from anon.models import AnonUser
+from asgiref.sync import async_to_sync
+from messages.models import Conversation, UserChannels
+from django.utils import timezone
+import datetime
+
+# __pragma__ ('skip')
+convo = 0
+# __pragma__ ('noskip')
+
+
+class ChatConsumer(WebsocketConsumer):
+    def connect(self):
+        self.accept()
+        self.user = None
+        self.groups = set()
+
+    def disconnect(self, close_code):
+        UserChannels.objects.filter(name=self.channel_name).delete()
+
+    def group_name(self, user):
+        return f"messaging-{self.user.id}"
+
+    def receive(self, text_data):
```

Comment:
Feels like you are doing a lot of things inside the function `receive` It might have been wise to refactor some business logic out.